anaconda_anon_usage/api.py

```diff
@@ -0,0 +1,130 @@
+# This module provides functions for reading and writing the
+# anonymous token set. It has been designed to rely only on the
+# Python standard library. In particular, hard dependencies on
+# conda must be avoided so that this package can be used in
+# child environments.
+
+import base64
+import functools
+import os
+import sys
+from collections import namedtuple
+from os.path import dirname, exists, expanduser, join
+from uuid import uuid4
+
+from . import __version__
+
+DEBUG = bool(os.environ.get("ANACONDA_ANON_USAGE_DEBUG"))
+Tokens = namedtuple("Tokens", ("version", "client", "session", "environment"))
+
+
+def _debug(s, *args, error=False):
+    if error or DEBUG:
+        print(s % args, file=sys.stderr)
+
+
+def _random_token():
+    return base64.urlsafe_b64encode(uuid4().bytes).strip(b"=").decode("ascii")
+
+
+def _saved_token(fpath, what):
+    """
+    Implements the saved token functionality. If the specified
+    file exists, and contains a token with the right format,
+    return it. Otherwise, generate a new one and save it in
+    this location. If that fails, return an empty string.
+    """
+    client_token = ""
+    _debug("%s token path: %s", what.capitalize(), fpath)
+    if exists(fpath):
+        try:
+            # Use just the first line of the file, if it exists
+            client_token = "".join(open(fpath).read().splitlines()[:1])
```

Comment:
Let's close this file properly to reduce the number of open files

```suggestion
            with open(fpath) as fp:
                client_token = "".join(fp.read().splitlines()[:1])
```